packages/libs/eda/src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -442,6 +451,85 @@ function HistogramViz(props: VisualizationProps<Options>) {
     [options, providedOverlayVariable, providedOverlayVariableDescriptor]
   );
 
+  // get distribution data
+  const subsettingClient = useSubsettingClient();
+
+  const getData = useCallback(async () => {
+    if (vizConfig.xAxisVariable != null && xAxisVariable != null) {
+      const [displayRangeMin, displayRangeMax, binWidth, binUnits] =
+        NumberVariable.is(xAxisVariable)
+          ? [
+              xAxisVariable.distributionDefaults.displayRangeMin ??
+                xAxisVariable.distributionDefaults.rangeMin,
+              xAxisVariable.distributionDefaults.displayRangeMax ??
+                xAxisVariable.distributionDefaults.rangeMax,
+              xAxisVariable.distributionDefaults.binWidth,
+              undefined,
+            ]
+          : [
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMin ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMin,
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMax ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMax,
+              (xAxisVariable as DateVariable).distributionDefaults.binWidth,
+              (xAxisVariable as DateVariable).distributionDefaults.binUnits,
+            ];
+
+      const distribution = await getDistribution<DistributionResponse>(
```

Comment:
I think it should be possible to call `subsettingClient.getDistribution()` directly, rather than as a callback to (same-named!) `getDistribution()`.  It will only return the "foreground" distribution, which is all we need.

I haven't tried this myself, but I assume it should be possible!